api/ipsets/validate

```diff
@@ -63,12 +64,21 @@ else:
     if status not in ['enabled', 'disabled']:
         invalid_attributes.append(invalid_attribute(status_p, "invalid"))
 
+
 # perform all other validations only if status == 'enabled'
 if status == 'enabled':
+    # geoipStatus
+    if (geoipStatus_p not in input_json) or (not input_json[geoipStatus_p]):
+        invalid_attributes.append(invalid_attribute(geoipStatus_p, "empty"))
+    else:
+        geoipStatus = input_json[geoipStatus_p]
+
+        if geoipStatus not in ['enabled', 'disabled']:
+            invalid_attributes.append(invalid_attribute(geoipStatus_p, "invalid"))
     # url
-    if (url_p not in input_json) or (not input_json[url_p]):
+    if (url_p not in input_json):
         invalid_attributes.append(invalid_attribute(url_p, "empty"))
-    else:
+    elif (input_json[url_p]):
```

Comment:
This change makes me think that URL is not mandatory anymore, but [inside download script](https://github.com/NethServer/nethserver-blacklist/blob/1353ae82bb0fbb2309aff196aac9b4eeee274f49/root/usr/share/nethserver-blacklist/download#L81) it looks like it is still mandatory. Am i missing something? :thinking:

Reply:
not a python guru like you @andre8244, I tested some ways. 

Now the url is not a mandatory, if the URL exists then we download (see https://github.com/NethServer/nethserver-blacklist/pull/37/files#diff-af724284fab7ff3dbe0e6c5d083dd81c6c30bddb4a030f75431ef294796b3404R38) same in the cron job ( see https://github.com/NethServer/nethserver-blacklist/pull/37/files#diff-78e4181675f2cbc15bc508d9a79c151096229bc2a3a4cb4bff0a29957c88b9e4R4)

Another way could be to have a third checkbox in the UI to enable git blacklist and a new property in the esmith key

we still check if the url is a valid url and if it is a git repository